### PR TITLE
Remove --genMapping

### PR DESCRIPTION
**Pull request by nc-x**
Closes https://github.com/nim-lang/Nim/issues/1592

### CODE REVIEW

File: compiler/commands.nim [hunk 1]
```diff
@@ -597,9 +597,6 @@ proc processSwitch*(switch, arg: string, pass: TCmdLinePass, info: TLineInfo;
   of "listcmd":
     expectNoArg(conf, switch, arg, pass, info)
     incl(conf.globalOptions, optListCmd)
-  of "genmapping":
-    expectNoArg(conf, switch, arg, pass, info)
```
Comment (Araq) [inline compiler/commands.nim]: Make it produce a warning "genmapping is obsolete and was ignored" or something like that.

**Comment by Araq** [comment]
Travis is red:

```
nim compile -f --symbolfiles:off --compileonly --gen_mapping --cc:gcc --skipUserCfg --os:windows --cpu:i386  compiler/nim.nim
command line(1, 2) Error: invalid command line option: '--gen_mapping'
```

**Comment by timotheecour** [comment]
@nc-x if we're gonna remove `--genMapping` (which, indeeed, doesn't seem to work), what should we use to get a mapping from mangled to demangled symbols?

**Comment by Araq** [comment]
It's not worth it. Let's do nothing instead.

**Comment by timotheecour** [comment]
@nc-x instead we can do following to fullfill use case of ppl who need a symbol mapping:
```
nim c --dump:foo.json main.nim
```
it'd share code with `nim dump` and produce relevant side information such as symbol mapping, but in a standard format (json).
(obviously, `nim dump` can't be used for that but code can be shared)

**Pull request closed without merging**